packages/typescript-eslint/src/config-helper.ts

```diff
@@ -85,11 +85,30 @@ export interface ConfigWithExtends extends TSESLint.FlatConfig.Config {
 export function config(
   ...configs: ConfigWithExtends[]
 ): TSESLint.FlatConfig.ConfigArray {
-  return configs.flatMap(configWithExtends => {
+  return configs.flatMap((configWithExtends, configIndex) => {
     const { extends: extendsArr, ...config } = configWithExtends;
     if (extendsArr == null || extendsArr.length === 0) {
       return config;
     }
+    const undefinedExtensions = extendsArr.reduce<number[]>(
+      (acc, extension, extensionIndex) => {
+        const maybeExtension = extension as
+          | TSESLint.FlatConfig.Config
+          | undefined;
+        if (!maybeExtension) {
+          acc.push(extensionIndex);
+        }
+        return acc;
+      },
+      [],
+    );
+    if (undefinedExtensions.length) {
+      throw new Error(
+        `Your config at index ${configIndex} contains undefined extensions ` +
```

Comment:
One thought - does it make sense to add the `name` if present? Like 
```suggestion
        `Your config at index ${configIndex} ${configWithExtends.name != null ? `, named '${configWithExtends.name}' : '(anonymous)'} contains undefined extensions ` +
```
or somesuch?